docs/frontend/testing/using-factories-to-generate-data-for-frontend-tests.md

```diff
@@ -0,0 +1,235 @@
+---
+sidebar_position: 8
+---
+
+# [WIP] Using factories to generate data for front-end tests
+
+## Writing a new factory
+
+Create and export a const called `{OBJECT_}FIELDS`. This constant should be an object whose keys are CAPITAL_SNAKE_CASE and whose values are the actual, camelCase names of the fields, e.g. `MY_FIELD: myField`. This makes using the factory easy, since the keys in question will be available to the user's IDE.
+
+```javascript
+  const MY_OBJECT_FIELDS = {
+    ID: 'id',
+    MY_FIELD: 'myField',
+  }
+```
+
+Create a const called `{object}Factory`, giving it a camelCase name. This const is a function that
+
+- takes params
+- returns a `baseFactory`, into which is passed an object containing
+  - `fields`: the fields your factory returns
+  - any other configuration passable to the `build()` functions [here](https://www.npmjs.com/package/@jackfranklin/test-data-bot), such as `postBuild` or `traits`.
+  - the spread params: `...params`.
+
+Export your factory by default.
+
+```javascript
+  const myObjectFactory = (params) => {
+    return baseFactory({
+      [BASE_FIELDS.FIELD]: {
+        [MY_OBJECT_FIELDS.ID]: '...',
+        [MY_OBJECT_FIELDS.MY_FIELD]: '...',
+      },
+      // optional postBuild and traits...,
+      ...params
+    })
+  }
+```
+
+### Build and post-build
+
+Factories generate data in two phases, the build and the post-build. Most values can be generated in the build phase. Use postBuild when the value of one field should be based on the value of another generated value.
+
+### Fields
+
+The `fields` object defines the structure and values of the object your factory will return. This happens at build time. Values of this object can be literals, functions, or another object containing either.
+
+```javascript
+  [BASE_OBJECT.FIELDS]: {
```

Comment:
**suggestion**: This is to match the implementation: https://github.com/transcom/mymove/pull/9233/files#diff-f220fcfa3edcfabb0300e0f877da7bf9790811e8ee9dabbd0d418cf0c80680f0R38-R44

```suggestion
  [BASE_FIELDS.FIELDS]: {
```